File: insert.py
#!/usr/bin/python
#Filename: insert.py

import sys

sArray = []
for i in range(1, len(sys.argv)):
    sArray.append(sys.argv[i])
print 'unsorted array', sArray

for i in range(1, len(sArray)):
    key = sArray[i]
    j = i-1
    while ((j>=0) and (key < sArray[j])):
        sArray[j+1] = sArray[j]
        j=j-1
    # in C, j will become -1 after a "j--", but in python it's not,
    # so here we don't use for loop
    sArray[j+1] = key
        
print 'sorted array', sArray

File: insert.c
#include <stdlib.h>
#include <stdio.h>

int main(int argc, char** argv)
{
    int i;
    int len = argc - 1;
    int array[len];
    if(argc < 2)
        return 0;
    /*read unsorted numbers*/
    for (i=0; i<len; i++)
    {
        array[i] = atoi(argv[i+1]);
    }
    printf("before insert sorted: ");
    for(i=0;i<len;i++)
    { 
        printf("%d ", array[i]); }
        printf("\n"); /*outer loop, from the 2nd number to the end*/ for(i=1; i<len; i++ ) { int j = i-1; int key = array[i]; /*inner loop, compare key[i] to numbers before it*/ /*after one inner loop, array[0] to array[i] become sequential*/ for(; j >= 0; j--)
        {
            if(key < array[j])
            {
                array[j+1] = array[j];
            }
            else
            {
                break;
            }
        }
        /* this is the correct spot of key[i]*/
        array[j+1] = key;
    }
    
    printf("after insert sorted: ");
    for(i=0;i<len;i++)
    {
        printf("%d ", array[i]);
    }
    printf(",O is N*N\n");
    return 1;
}
